File: bank-extract-api/src/main/java/br/com/bank/bankextractapi/domain/Activity.java
package br.com.bank.bankextractapi.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Activity {

    public enum ActivityType {
        DEPOSIT, WITHDRAW, TRANSFER
    }

    private Long id;
    private String sourceAccount;
    private String targetAccount;
    private ActivityType type;
    private BigDecimal value;
    private BigDecimal balance;
    private LocalDateTime date;
}

File: bank-api/src/main/kotlin/br/com/bank/domain/Activity.kt
package br.com.bank.domain

import java.math.BigDecimal
import java.time.LocalDateTime
import javax.validation.constraints.Min

open class Activity(val id: Long?,
                    val ownerAccount: Account,
                    val sourceAccount: Account?,
                    val targetAccount: Account?,
                    val type: TypeActivity,

                    @Min(1)
                    val value: BigDecimal,
                    val balance: BigDecimal,
                    val date: LocalDateTime) {

    constructor(ownerAccount: Account,
                sourceAccount: Account?,
                targetAccount: Account?,
                type: TypeActivity,
                value: BigDecimal,
                balance: BigDecimal,
                date: LocalDateTime): this(null, ownerAccount, sourceAccount, targetAccount, type, value, balance, date)

    constructor(ownerAccount: Account,
                type: TypeActivity,
                value: BigDecimal,
                balance: BigDecimal,
                date: LocalDateTime): this(null, ownerAccount, null, null, type, value, balance, date)


}